crates/fj-core/src/objects/object_set.rs

```diff
@@ -260,4 +267,38 @@ mod tests {
         // Nothing more to test. `ObjectSet` panics on duplicate objects, and it
         // shouldn't do that in this case.
     }
+
+    #[test]
+    fn deduplicate_with_hashset() {
+        let mut core = Core::new();
+
+        let bare_cycle = Cycle::new([]);
+        let cycle = bare_cycle.insert(&mut core);
+
+        let standard_set = HashSet::from([cycle.clone(), cycle.clone()]);
+
+        assert_eq!(standard_set, [cycle].into());
+    }
```

Comment:
I don't think this belongs here. It doesn't actually test `ObjectSet`. I'd suggest either removing it, or adapting it as an example in the documentation of `ObjectSet::new`.

I'm going to merge the pull request anyway, as I don't think this is a big deal. Feel free to send a follow-up PR, if you want to. If not, I'll most likely remove it in a future cleanup pass.